File: src/main/java/io/primeaspect/calculator/controller/StatisticsController.java
package io.primeaspect.calculator.controller;

import io.primeaspect.calculator.dto.response.ExpressionListResponse;
import io.primeaspect.calculator.model.Number;
import io.primeaspect.calculator.service.StatisticsService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/statistics")
public class StatisticsController {
    private final StatisticsService service;

    public StatisticsController(StatisticsService service) {
        this.service = service;
    }

    @RequestMapping(value = "/date", params = "date")
    public ExpressionListResponse findAllByDate(@RequestParam String date) {
        return service.findAllByDate(date);
    }

    @RequestMapping(value = "/operation", params = "operation")
    public ExpressionListResponse findAllByOperation(@RequestParam String operation) {
        return service.findAllByOperation(operation);
    }

    @RequestMapping(value = "/count/date", params = "date")
    public int getCountByDate(@RequestParam String date) {
        return service.getCountByDate(date);
    }

    @RequestMapping(value = "/count/operation", params = "operation")
    public int getCountByOperation(@RequestParam String operation) {
        return service.getCountByOperation(operation);
    }

    @GetMapping("/popular")
    public Number getPopularNumber() {
        return service.getPopularNumber();
    }
}

File: src/main/kotlin/io/primeaspect/calculator/controller/StatisticsController.kt
package io.primeaspect.calculator.controller

import io.primeaspect.calculator.service.StatisticsService
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDate

@RestController
@RequestMapping("/statistics")
class StatisticsController(private var service: StatisticsService) {
    @GetMapping(value = ["/date"], params = ["date"])
    fun findAllByDate(
            @RequestParam
            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
            date: LocalDate
    ) = service.findAllByDate(date)

    @GetMapping(value = ["/operation"], params = ["operation"])
    fun findAllByOperation(
            @RequestParam
            operation: String
    ) = service.findAllByOperation(operation)

    @GetMapping(value = ["/count/date"], params = ["date"])
    fun getCountByDate(
            @RequestParam
            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
            date: LocalDate
    ) = service.getCountByDate(date)

    @GetMapping(value = ["/count/operation"], params = ["operation"])
    fun getCountByOperation(
            @RequestParam
            operation: String
    ) = service.getCountByOperation(operation)

    @GetMapping("/popular")
    fun getPopularNumber() = service.getPopularNumber()

    @GetMapping(value = ["/popular/date"], params = ["date"])
    fun getPopularNumberByDate(
            @RequestParam
            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
            date: LocalDate
    ) = service.getPopularNumberByDate(date)
}
